server/sync_watcher.go

```diff
@@ -272,13 +272,8 @@ func (watcher *SyncWatcher) processSyncWatch(ctx context.Context, path string) e
 				if response.Err != nil {
 					return response.Err
 				}
+				watcher.watchExtensionHandler(response)
 
-				watcher.queue.Add(job.NewJob(
```

Comment:
I was hoping this could be achieved by configuration change (parallelism settings per watcher or sth). Maybe we shouldn't remove the whole feature.